lib/danger/scm_source/git_repo.rb

```diff
@@ -4,20 +4,25 @@
 
 module Danger
   class GitRepo
-    attr_accessor :diff, :log
+    attr_accessor :diff, :log, :folder
 
     def diff_for_folder(folder, from: "master", to: "HEAD")
-      repo = Git.open folder
+      self.folder = folder
+      repo = Git.open self.folder
 
       merge_base = repo.merge_base(from, to)
+      ensure_merge_base_exists!(merge_base)
```

Comment:
interesting, in my case https://github.com/danger/danger.systems/pull/102 / https://circleci.com/gh/danger/danger.systems/89 the commit isn't in the list of commits at all

I don't think it's unreasonable to have the same checks for both `from` and `to`, then the `merge_base`